streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -443,13 +443,12 @@ private void to(final TopicNameExtractor<K, V> topicExtractor, final ProducedInt
         addGraphNode(sinkNode);
     }
 
-    @Override
-    public <K1, V1> KStream<K1, V1> transform(final TransformerSupplier<? super K, ? super V, KeyValue<K1, V1>> transformerSupplier,
-                                              final String... stateStoreNames) {
-        Objects.requireNonNull(transformerSupplier, "transformerSupplier can't be null");
+    private <K1, V1> KStream<K1, V1> doTransform(final ProcessorSupplier processorSupplierForGraph,
```

Comment:
Do we really need two distinct objects for the processor supplier? This is an intermediate solution, isn't it?

Reply:
@guozhangwang This code and comment is outdated. The method `doTransform()` takes one supplier and is not modified by this PR any longer.